backend/src/main/java/org/cryptomator/hub/api/SettingsResource.java

```diff
@@ -0,0 +1,62 @@
+package org.cryptomator.hub.api;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Max;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.cryptomator.hub.entities.Settings;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+
+@Path("/settings")
+public class SettingsResource {
+
+	@Inject
+	Settings.Repository settingsRepo;
+
+	@GET
+	@RolesAllowed("user")
+	@Produces(MediaType.APPLICATION_JSON)
+	@Operation(summary = "get the billing information")
+	@APIResponse(responseCode = "200")
+	@Transactional
+	public SettingsDto get() {
+		return SettingsDto.fromEntity(settingsRepo.get());
+	}
+
+	@PUT
+	@RolesAllowed("admin")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Operation(summary = "update settings")
+	@APIResponse(responseCode = "204", description = "token set")
+	@APIResponse(responseCode = "400", description = "invalid settings")
+	@APIResponse(responseCode = "403", description = "only admins are allowed to update settings")
+	@Transactional
+	public Response put(@NotNull @Valid SettingsDto dto) {
+		var settings = settingsRepo.get();
+		settings.setWotMaxDepth(dto.wotMaxDepth);
+		settings.setWotIdVerifyLen(dto.wotIdVerifyLen);
+		settingsRepo.persist(settings);
+		return Response.status(Response.Status.NO_CONTENT).build();
+	}
+
+	public record SettingsDto(@JsonProperty("hubId") String hubId, @JsonProperty("wotMaxDepth") @Min(0) @Max(10) int wotMaxDepth, @JsonProperty("wotIdVerifyLen") @Min(0) int wotIdVerifyLen) {
```

Comment:
The current max of `wotMaxDepth` is 9, see https://github.com/cryptomator/hub/pull/281/files#diff-8237ab834f761b84d37b515e3bca1a1c40bf6821d87db8b39455f2c47bc59fbbR3

```suggestion
	public record SettingsDto(@JsonProperty("hubId") String hubId, @JsonProperty("wotMaxDepth") @Min(0) @Max(9) int wotMaxDepth, @JsonProperty("wotIdVerifyLen") @Min(0) int wotIdVerifyLen) {
```